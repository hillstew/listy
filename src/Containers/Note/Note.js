import React from 'react';

const Note = () => {


  return (
    <div className='note-card'>

    </div>
  )
}

export default Note;