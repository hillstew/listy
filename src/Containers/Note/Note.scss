h3 {
  font-family: $main-font;
  margin: 10px;
  margin-bottom: 20px;
  font-size: 30px;
  font-weight: 400;
}

i:hover {
  color: $main-green;
}

ul {
  font-family: $main-font;
  list-style-type: none;
  padding-inline-start: 0;
  margin: 10px;
  font-size: 18px;
}

.completed-list {
  margin-top: 30px;
  color: $main-grey;
  border-top: 1px solid $main-grey;
  padding-top: 10px;
}

.check-box {
  display: inline-block;
  height: 16px;
  width: 16px;
  margin-right: 15px;
  background-color: $check-box-color;
  box-shadow: 2px 2px 5px $main-grey;
  cursor: pointer;

  &:hover {
    box-shadow: 0 2px 5px $main-grey;
  }
}

.completed {
  background-color: $main-light-grey;
  background: url('./images/check.png');
  background-size: cover;
  box-shadow: none;
}

.note-card {
  width: 280px;
  height: 400px;
  background-color: #FFF;
  border: 1px solid $main-grey;
  border-radius: 10px;
  box-shadow: 2px 2px 8px black;
  padding: 20px;
}

.edit-button-note {
  background: url('./images/pencil.png');
  background-size: cover;
  border: none;
  cursor: pointer;
  height: 20px;
  margin: 0 auto;
  transition-duration: 1s;
  width: 20px;

  &:hover {
    transform: scale(1.3);
  }
}

a {
  text-decoration: none;
}

a:visited {
  color: black;
}

li {
  z-index: 20;
}