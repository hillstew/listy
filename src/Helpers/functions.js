export const getIndex = (id, issues) => {
    return issues.findIndex((issue) => issue.id == id);
};

export const createIssuesCopy = (issues) => {
    return issues.slice();
};