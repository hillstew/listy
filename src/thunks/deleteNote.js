import { setLoading, setError, removeNote } from '../actions';
import API from '../utils/api';

export const deleteNote = (id) => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const result = await API.fetchData(`notes/${id}`, 'DELETE');
      dispatch(setLoading(false));
      dispatch(removeNote(result));
    } catch (error) {
      dispatch(setError(error));
    }
  }
}