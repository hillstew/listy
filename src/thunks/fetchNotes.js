import { setLoading, setError, setNotes } from '../actions'

export const fetchNotes = () => {
  const url = 'http://localhost:3001/api/v1/notes';
  return async (dispatch) =>  {
    try {
      dispatch(setLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(setLoading(false))
      const notes = await response.json()
      dispatch(setNotes(notes))
    } catch (error) {
      dispatch(setError(error.message))
    }
  }
}