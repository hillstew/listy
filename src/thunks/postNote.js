import { setLoading, setError, addNote, togglePopup } from '../actions';
import API from '../utils/api';

export const postNote = (note) => {
  const { title, issues } = note;
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const result = await API.fetchData('notes', 'POST', { title, issues });
      dispatch(setLoading(false));
      dispatch(addNote(result));
    } catch (error) {
      dispatch(setError(error));
    }
  }
}