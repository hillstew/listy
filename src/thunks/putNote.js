import { setLoading, setError, updateNote } from '../actions';
import API from '../utils/api';

export const putNote = (note) => {
  const { id, title, issues } = note;
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      await API.fetchData(`notes/${id}`, 'PUT', { title, issues });
      dispatch(updateNote(note));
    } catch (error) {
      dispatch(setError(error));
    }
    dispatch(setLoading(false));
  }
}