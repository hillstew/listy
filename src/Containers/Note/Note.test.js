import React from 'react'
import { shallow } from 'enzyme';
import { Note, mapDispatchToProps } from './Note';
import { putNote } from '../../thunks/putNote';
import { getIndex, createIssuesCopy } from '../../Helpers/functions';

jest.mock('../../thunks/putNote')
jest.mock('../../Helpers/functions')

describe('Note', () => {
    describe('Note component', () => {
        let wrapper;
        let mockNote;
        let mockEvent;
        let mockPutNote;

        beforeEach(() => {
            mockEvent = { target: { 
                id: '1',
                value: 'new body'
            }};
            mockNote = {
                id: '1',
                title: 'Title',
                color: 'white',
                issues: [
                    { id: '1', body: 'issue text', completed: false },
                    { id: '2', body: 'more issue text', completed: true },
                    { id: '3', body: 'text', completed: true }
                ]
            }
            let mockKey = 1;
            mockPutNote = jest.fn();
            wrapper = shallow(<Note note={mockNote} key={mockKey} putNote={mockPutNote}/>)
            getIndex.mockReturnValue('1');
            createIssuesCopy.mockReturnValue(mockNote.issues);
        });

        it('should match the correct snapshot', () => {
            expect(wrapper).toMatchSnapshot()
        });

        it('should call getIndex with the correct params', () => {
            wrapper.instance().toggleIssueCompletion(mockEvent);
            expect(getIndex).toHaveBeenCalledWith('1', mockNote.issues);
        });

        it('should call createIssuesCopy with the correct params', () => {
            wrapper.instance().toggleIssueCompletion(mockEvent);
            expect(createIssuesCopy).toHaveBeenCalledWith(mockNote.issues);
        });

        it('should call putNote with the correct params', async () => {
            wrapper.instance().toggleIssueCompletion(mockEvent);
            await expect(wrapper.instance().props.putNote).toHaveBeenCalledWith(mockNote);
        });

        it('renderIssues should filter complete/incomplete issues of array', () => {
            const completeIssues = wrapper.instance().renderIssues(true);
            expect(completeIssues.length).toEqual(2);
            
            const incompleteIssues = wrapper.instance().renderIssues(false);
            expect(incompleteIssues.length).toEqual(1);
        });
    });

    describe('mapDispatchToProps', () => {
        it('should call dispatch with a putNote action when putNote is called', () => {
            let mockNote = {
                title: 'Title',
                issues: [],
            }
            const mockDispatch = jest.fn();
            const actionToDispatch = putNote(mockNote);
            const mappedProps = mapDispatchToProps(mockDispatch);
            mappedProps.putNote(mockNote);
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
        });
    });
});