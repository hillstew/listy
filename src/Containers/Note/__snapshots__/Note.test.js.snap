// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Note component should match the correct snapshot 1`] = `
<div
  className="note-card undefined"
>
  <h3>
    Title
  </h3>
  <ul
    className="incomplete-list"
  />
  <ul
    className="completed-list"
  />
  <Link
    replace={false}
    to="/notes/undefined"
  >
    <button
      className="edit-button-note"
    />
  </Link>
</div>
`;
